src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -96,8 +99,14 @@ private IEnumerable<UserDefinedFunction> CreateUserDefinedFunctions(IEnumerable<
                 {
                     continue;
                 }
-                
-                var func = new UserDefinedFunction(udfName.Value, udf.ReturnType.GetFormulaType()._type, udf.Body, udf.IsImperative, udf.Args);
```

Comment:
>GetFormulaType [](http://example.com/codeflow?start=81&length=14)

Why not fix GetFormulaType to return valid type here?
Why not update String FormulaType to Text?

Reply:
Modified GetFormulaType() to return valid type